runtime/src/main/java/org/corfudb/protocols/wireprotocol/CorfuMsgType.java

```diff
@@ -82,41 +84,42 @@
     ERROR_VALUE_ADOPTED(59,new TypeToken<CorfuPayloadMsg<ReadResponse>>() {}),
 
     // EXTRA CODES
-    LAYOUT_ALREADY_BOOTSTRAP(60, TypeToken.of(CorfuMsg.class), true),
-    LAYOUT_PREPARE_ACK(61, new TypeToken<CorfuPayloadMsg<LayoutPrepareResponse>>(){}, true),
-    RESTART(62, TypeToken.of(CorfuMsg.class), true),
-    KEEP_ALIVE(63, TypeToken.of(CorfuMsg.class), true),
+    LAYOUT_ALREADY_BOOTSTRAP(60, TypeToken.of(CorfuMsg.class), true, false),
+    LAYOUT_PREPARE_ACK(61, new TypeToken<CorfuPayloadMsg<LayoutPrepareResponse>>(){}, true, false),
+    RESTART(62, TypeToken.of(CorfuMsg.class), true, false),
+    KEEP_ALIVE(63, TypeToken.of(CorfuMsg.class), true, true),
```

Comment:
I think for KEEP_ALIVE, it's better not to ignore cluster ID, if a client is from a different cluster, it's unnecessary for the servers in this cluster to keep alive the channel to that client.

Reply:
The reason for keep alive ignoring the clusterId checks is because there is a possibility of sending a keep alive messages after the handshake is completed but before the cluster was formed. Basically this scenario might occur:

1. Handshake occurs to establish a trust (making sure a client and a server have correct certificates). We are not concerned with cluster level here yet because there is no cluster per se. 
2. Channel connection is established after a successful handshake. 
3. Channel is idle, and a client has not bootstrapped a cluster yet. A keep alive message is sent to the server router. If we did not ignore, this would have thrown an exception because of the clusterId check but there was never a cluster in the first place, we just established the connection.